src/web/mod.rs

```diff
@@ -307,13 +302,14 @@ fn match_version(conn: &Connection, name: &str, version: Option<&str>) -> Option
         versions_sem
     };
 
-    for version in &versions_sem {
-        if req_sem_ver.matches(&version.0) {
-            return Some(MatchVersion {
-                corrected_name,
-                version: MatchSemver::Semver((version.0.to_string(), version.1)),
-            });
-        }
+    if let Some((version, id)) = versions_sem
+        .iter()
+        .find(|(vers, _)| req_sem_ver.matches(vers))
```

Comment:
What's the reason for this change? I think it was more readable before.

Reply:
We have potentially better optimization from iterators and it's more idiomatic